exporter/datadogexporter/internal/clientutil/http.go

```diff
@@ -29,8 +30,14 @@ var (
 
 // NewHTTPClient returns a http.Client configured with a subset of the confighttp.ClientConfig options.
 func NewHTTPClient(hcs confighttp.ClientConfig) *http.Client {
+	// If the ProxyURL field in the configuration is set, the HTTP client will use the proxy.
+	// Otherwise, the HTTP client will use the system's proxy settings.
+	httpProxy := http.ProxyFromEnvironment
+	if parsedProxyURL, err := url.Parse(hcs.ProxyURL); err == nil && parsedProxyURL.Scheme != "" {
+		httpProxy = http.ProxyURL(parsedProxyURL)
+	}
 	transport := http.Transport{
-		Proxy: http.ProxyFromEnvironment,
+		Proxy: httpProxy,
```

Comment:
What's the benefit of introducing this instead of using [the environment variables checked by `http.ProxyFromEnvironment` ](https://pkg.go.dev/net/http#ProxyFromEnvironment)?

Reply:
We are using this in conjunction with `cmd/opampsupervisor`. As part of that, it's much easier for us to manage our DD proxy via configuration that can be sent from a server to agents.

`cmd/opampsupervisor` does not allow you to control the environment variables where you launch the collector, whereas it does allow you extensive control of the configuration that you pass to it.

we have a highly restricted network, and we may need to change the DD proxy configuration, so that's why.

Reply:
Thanks for the explanation, that seems like a valid reason to me to add this